File: app/src/main/java/com/irontec/axier/mvvmexample/MainActivity.java
package com.irontec.axier.mvvmexample;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.irontec.axier.mvvmexample.adapters.ArticleAdapter;
import com.irontec.axier.mvvmexample.databinding.ActivityMainBinding;
import com.irontec.axier.mvvmexample.mock.ArticleMockHelper;
import com.irontec.axier.mvvmexample.model.Article;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ActivityMainBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        binding.contactList.setLayoutManager(layoutManager);

        List<Article> articles = ArticleMockHelper.getMockArticleData();

        ArticleAdapter adapter = new ArticleAdapter(articles);
        binding.contactList.setAdapter(adapter);
    }
}

File: app/src/main/java/com/irontec/axier/mvvmexample/view/MainActivity.kt
package com.irontec.axier.mvvmexample.view

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.LinearLayoutManager
import com.irontec.axier.mvvmexample.R
import com.irontec.axier.mvvmexample.databinding.ActivityMainBinding
import com.irontec.axier.mvvmexample.mock.ArticleMockHelper

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)
        val layoutManager = LinearLayoutManager(this)
        binding.contactList.layoutManager = layoutManager

        val articles = ArticleMockHelper.mockArticleData

        val adapter = ArticleAdapter(articles)
        binding.contactList.adapter = adapter
    }
}
